### PR TITLE
add a task to the todo list

### PR DESCRIPTION
**Pull request by anshikaa12**

### CODE REVIEW

File: src/components/EditModal.jsx [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import React from "react";
-import { useToggle } from "../contexts/editContext";
+import { useTask, useTodo, useToggle } from "../contexts";
 
 function EditModal() {
   const { active, setActive } = useToggle();
+  const { todoState, todoDispatch } = useTodo();
+  const { taskDispatch } = useTask();
   function submitTaskHandler() {
     setActive(!active);
```
Comment (deepak29-git) [inline src/components/EditModal.jsx]: suggestion: you should use es6 syntax

**Review by deepak29-git — APPROVED** [review]
everything looks good and code is clean

File: src/components/EditModal.jsx (hunk 1 above)
Comment (abhij1607) [inline src/components/EditModal.jsx]: use functional update here

File: src/components/TodoList.jsx [hunk 2]
```diff
@@ -1,42 +1,36 @@
 import React from "react";
-import { useToggle } from "../contexts/editContext";
+import { useToggle, useTask } from "../contexts";
 
 function TodoList() {
   const { active, setActive } = useToggle();
   function editTaskHandler() {
     setActive(!active);
```
Comment (abhij1607) [inline src/components/TodoList.jsx:7]: impure update right now, use functional update

**Review by abhij1607 — COMMENTED** [review]
Good folder structure | | Very neat & clean code | | All good practices followed

Left a suggestion in comment

**Review by abhij1607 — APPROVED** [review]
approving the changes, good to merge

**Pull request merged**